src/test/java/com/thebois/utils/MatrixUtilsTests.java

```diff
@@ -64,22 +61,18 @@ public void forEachElementGetsAllElementsInOrder(final Integer[][] matrix,
 
     @ParameterizedTest
     @MethodSource("getMatricesToTest")
-    public void matrixToCollectionTest(final Optional<String>[][] matrix,
+    public void matrixToCollection(final String[][] matrix,
                                        final Collection<String> expectedElements) {
-        // Arrange
-
         // Act
-        final Collection<String> stringCollection = MatrixUtils.matrixToCollection(matrix);
+        final Collection<String> stringCollection = MatrixUtils.toCollection(matrix);
 
         // Assert
         assertThat(stringCollection).isEqualTo(expectedElements);
     }
 
     private static Stream<Arguments> getCorrectCoordinatesToTest() {
-        final Optional<String>[][] stringOptionalMatrix = new Optional[][] {
-            { Optional.of("1"), Optional.empty(), Optional.of("3") },
-            { Optional.empty(), Optional.empty(), Optional.of("6") },
-            { Optional.of("7"), Optional.of("8"), Optional.of("9") },
+        final String[][] stringOptionalMatrix = new String[][] {
```

Comment:
style

`stringOptionalMatrix` -> `stringMatrix`

And isn't there a better name? `stringMatrix` tells me nothing at all about what it represents. It's hard to parse out what it is supposed to contain without reading the entire file.